### ISSUE TITLE
undefined method `split' for nil:NilClass in 3.0.8 (rails 5)

### ISSUE THREAD

**Issue by jeppeliisberg** [opbeat/opbeat-ruby#35]
I see this in my production log after upgrading to opbeat 3.0.8 (Rails 5)

I'll take a closer look asap, but wanted to file the issue here fast as well so others can chip in ;-)

```
Aug 26 09:05:06 myphoner app/web.2:  F, [2016-08-26T07:05:06.483114 #134] FATAL -- : vendor/bundle/ruby/2.3.0/gems/opbeat-3.0.8/lib/opbeat/middleware.rb:10:in `call' 
Aug 26 09:26:19 myphoner app/web.2:  E, [2016-08-26T07:26:19.663950 #134] ERROR -- : ** [Opbeat] Error capturing or sending exception undefined method `split' for nil:NilClass 
```

**Comment by jeppeliisberg** [comment]
Examining this error, I now see another error in my local environment:

```
** [Opbeat] Failed POST: #<Opbeat::Error: Error from Opbeat server (400): {
    "status": 400, 
    "structured_error": {
        "http": {
            "url": "must be the full URL, including scheme and host"
        }
    }, 
    "error_message": "Unable to validate data."
}>
```

I'm guessing the `headers` fix in https://github.com/opbeat/opbeat-ruby/issues/34 was not enough - maybe there's some more parameters missing now?

**Comment by jeppeliisberg** [comment]
@mikker when debugging, I can see that the http object is not valid - maybe the format of the headers changed in rails 5?

From: /lib/opbeat/error_message/http.rb @ line 54 Opbeat::ErrorMessage::HTTP.from_rack_env:

```
    50:         if filter = opts[:filter]
    51:           http.apply_filter filter
    52:         end
    53: 
 => 54:         binding.pry_remote
    55: 
    56:         http
    57:       end
    58: 
    59:       def apply_filter filter

[1] pry(Opbeat::ErrorMessage::HTTP)> http.url
=> "://::0"
```

**Comment by mikker** [comment]
@jeppeliisberg Can you check if this PR fixes your problem? https://github.com/opbeat/opbeat-ruby/issues/36

**Comment by mikker** [comment]
Fixed in 8ac500c52ce43624a7cd6b615a8ea151338dd051
